protocol/x/clob/keeper/liquidations.go

```diff
@@ -138,8 +138,13 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
 	for i := 0; i < int(k.Flags.MaxDeleveragingAttemptsPerBlock) && i < len(unfilledLiquidations); i++ {
 		liquidationOrder := unfilledLiquidations[i]
 
-		subaccountId := liquidationOrder.GetSubaccountId()
 		perpetualId := liquidationOrder.MustGetLiquidatedPerpetualId()
+		subaccountId := liquidationOrder.GetSubaccountId()
+
+		subaccount := k.subaccountsKeeper.GetSubaccount(ctx, subaccountId)
+		if _, exists := subaccount.GetPerpetualPositionForId(perpetualId); !exists {
+			continue
```

Comment:
Is it worth statting how often this happens? Up to you, hopefully deleveraging is pretty rare so this doesn't happen.